libbeat/esleg/eslegclient/connection.go

```diff
@@ -110,6 +113,14 @@ func NewConnection(s ConnectionSettings) (*Connection, error) {
 		}
 	}
 
+	name, err := os.Executable()
```

Comment:
Perhaps `UserAgent` could be a setting within `ConnectionSettings` that is passed in by the caller. That seems like something that it could pass in when creating the ES client. Like this location where the Beat name is known https://github.com/elastic/beats/blob/04762fd7d46fce4389b94fd94a7629cc635f1557/libbeat/outputs/elasticsearch/elasticsearch.go#L93.